src/migration.types.ts

```diff
@@ -85,13 +85,13 @@ export interface ErrorReport {
  * A function that can be called to migrate a dashboard from one version to another.
  * It is called within {@link produce | https://immerjs.github.io/immer/produce}, so it can safely mutate its input `dashboardState`.
  */
-export type MigrateDashboardCallback<FromDashboardState, ToDashboardState> = (
+export type MigrateDashboardCallback<
+  FromDashboardState,
+  ToDashboardState,
+  Options extends Record<string, unknown> = Record<string, unknown>,
+> = (
   dashboardState: FromDashboardState,
-  {
-    dataModels,
-    keysOfWidgetPluginsToRemove,
-    onErrorWhileMigratingWidget,
-  }: {
+  args: {
```

Comment:
```suggestion
  options: {
```
It's not a huge deal, but I'm not a fan of `args`: `dashboardState` is also an argument, and the main one actually. To me those are more "options". However the type is called `Options` already.
We could:
- rename `args` in `options` and the type in `ExtraOptions`
- rename `args` in `baseOptions` and keep the type as `Options`
- rename `args` in `baseOptions` and the type in `ExtraOptions`

I prefer the first option, but if you disagree feel free to go for another option (no pun intended lmao), again I don't think it's a big deal.

Reply:
What I don't like in  "options" is that `dataModels` is not optional for example.
Whereas things like `namesOfCalculatedMeasuresToMigrate` is an opt-in of 5.0 of 5.1, but it's not there for every migration step.
That's why I went for `args` for the whole object but `Options` for those extra opt-in things.

Reply:
Fair enough, it's a good point. Let's go with `args`!